--- build/Build/MDD.pm	2012-11-26 15:06:19.000000000 +0000
+++ build/Build/MDD.pm.new	2014-01-13 11:09:13.772315088 +0000
@@ -63,6 +63,7 @@
     $destdir    = "" unless $destdir;
     my $confdir = $destdir . $self->install_destination('conf');
     unless (-e "$confdir/mdd.conf") {
+        mkdir "$confdir";
         print "cp conf/mdd.conf -> $confdir/mdd.conf\n";
         copy('conf/mdd.conf', "$confdir/mdd.conf") or die "$!\n";
         chmod(0644, "$confdir/mdd.conf")
@@ -113,9 +114,9 @@
         copy('init/gentoo', $init);
         chmod(0755, $init);
         if ($new) {
-            $self->do_system('rc-update add mdd default');
+            #$self->do_system('rc-update add mdd default');
         }
-        $self->do_system("$init restart");
+        #$self->do_system("$init restart");
         return;
     }
 
@@ -140,7 +141,7 @@
 sub ACTION_install {
     my $self = shift;
     croak("$0 must be run as root to install") unless ($> == 0);
-    $self->create_account();
+    #$self->create_account();
     $self->SUPER::ACTION_install;
     $self->install_conf();
     if ($self->feature('backend')) {
