diff -ruN mythtv-0.28/mythplugins/configure mythtv-0.28_new/mythplugins/configure
--- mythtv-0.28/mythplugins/configure	2016-04-11 20:30:28.000000000 +0100
+++ mythtv-0.28_new/mythplugins/configure	2016-11-12 17:12:58.280078053 +0000
@@ -682,7 +682,7 @@
         echo "MythMusic requires FLAC."
     fi
 
-    if ! check_lib cdio/cdio.h cdio_open -lcdio || ! check_lib cdio/cdda.h cdio_cddap_open -lcdio_cdda || ! check_lib cdio/paranoia.h cdio_paranoia_init -lcdio_paranoia ; then
+    if ! check_lib cdio/cdio.h cdio_open -lcdio || (! check_lib cdio/cdda.h cdio_cddap_open -lcdio_cdda && ! check_lib cdio/paranoia/cdda.h cdio_cddap_open -lcdio_cdda) || (! check_lib cdio/paranoia.h cdio_paranoia_init -lcdio_paranoia && ! check_lib cdio/paranoia/paranoia.h cdio_paranoia_init -lcdio_paranoia) ; then
         disable cdio
     fi
 
@@ -901,11 +901,16 @@
     fi
 
     if enabled cdio ; then
-      echo "#define HAVE_CDIO 1" >> ./mythmusic/mythmusic/config.h
+      if check_lib cdio/cdda.h cdio_cddap_open -lcdio_cdda ; then
+        echo "#define HAVE_CDIO 1" >> ./mythmusic/mythmusic/config.h
+      else
+        echo "#define HAVE_CDIO_PARANOIA 1" >> ./mythmusic/mythmusic/config.h
+      fi
       echo "CONFIG += cdio" >> ./mythmusic/mythmusic/config.pro
       echo "        libcdio        support will be included in MythMusic"
     else
       echo "#undef  HAVE_CDIO" >> ./mythmusic/mythmusic/config.h
+      echo "#undef  HAVE_CDIO_PARANOIA" >> ./mythmusic/mythmusic/config.h
       echo "        libcdio        support will not be included in MythMusic"
     fi
 
diff -ruN mythtv-0.28/mythplugins/mythmusic/mythmusic/cddecoder.cpp mythtv-0.28_new/mythplugins/mythmusic/mythmusic/cddecoder.cpp
--- mythtv-0.28/mythplugins/mythmusic/mythmusic/cddecoder.cpp	2016-04-11 20:30:28.000000000 +0100
+++ mythtv-0.28_new/mythplugins/mythmusic/mythmusic/cddecoder.cpp	2016-11-12 17:14:36.559380936 +0000
@@ -14,7 +14,11 @@
 #include <QString>
 
 // libcdio
-#include <cdio/cdda.h>
+#ifdef HAVE_CDIO
+# include <cdio/cdda.h>
+#else
+# include <cdio/paranoia/cdda.h>
+#endif
 #include <cdio/logging.h>
 
 // MythTV
diff -ruN mythtv-0.28/mythplugins/mythmusic/mythmusic/cddecoder.h mythtv-0.28_new/mythplugins/mythmusic/mythmusic/cddecoder.h
--- mythtv-0.28/mythplugins/mythmusic/mythmusic/cddecoder.h	2016-04-11 20:30:28.000000000 +0100
+++ mythtv-0.28_new/mythplugins/mythmusic/mythmusic/cddecoder.h	2016-11-12 17:12:58.331077172 +0000
@@ -15,6 +15,11 @@
 # include <cdio/paranoia.h>
 #endif
 
+#ifdef HAVE_CDIO_PARANOIA
+# include <cdio/paranoia/cdda.h>
+# include <cdio/paranoia/paranoia.h>
+#endif
+
 class MusicMetadata;
 
 class CdDecoder : public Decoder
@@ -82,7 +87,7 @@
     int m_settracknum;
     int m_tracknum;
 
-#ifdef HAVE_CDIO
+#if defined(HAVE_CDIO) || defined(HAVE_CDIO_PARANOIA)
     CdIo_t *m_cdio;
     cdrom_drive_t *m_device;
     cdrom_paranoia_t *m_paranoia;
diff -ruN mythtv-0.28/mythplugins/mythmusic/mythmusic/cdrip.cpp mythtv-0.28_new/mythplugins/mythmusic/mythmusic/cdrip.cpp
--- mythtv-0.28/mythplugins/mythmusic/mythmusic/cdrip.cpp	2016-04-11 20:30:28.000000000 +0100
+++ mythtv-0.28_new/mythplugins/mythmusic/mythmusic/cdrip.cpp	2016-11-12 17:12:58.338077051 +0000
@@ -12,6 +12,11 @@
 # include <cdio/paranoia.h>
 #endif //def HAVE_CDIO
 
+#ifdef HAVE_CDIO_PARANOIA
+# include <cdio/paranoia/cdda.h>
+# include <cdio/paranoia/paranoia.h>
+#endif
+
 // C++ includes
 #include <iostream>
 #include <memory>
@@ -52,7 +57,7 @@
 
 // MythMusic includes
 #include "cdrip.h"
-#ifdef HAVE_CDIO
+#if defined(HAVE_CDIO) || defined(HAVE_CDIO_PARANOIA)
 #include "cddecoder.h"
 #endif // HAVE_CDIO
 #include "encoder.h"
@@ -64,7 +69,7 @@
 #include "mythlogging.h"
 #include "musicutils.h"
 
-#ifdef HAVE_CDIO
+#if defined(HAVE_CDIO) || defined(HAVE_CDIO_PARANOIA)
 // libparanoia compatibility
 #ifndef cdrom_paranoia
 #define cdrom_paranoia cdrom_paranoia_t
@@ -132,7 +137,7 @@
 
 static long int getSectorCount (QString &cddevice, int tracknum)
 {
-#ifdef HAVE_CDIO
+#if defined(HAVE_CDIO) || defined(HAVE_CDIO_PARANOIA)
     QByteArray devname = cddevice.toLatin1();
     cdrom_drive *device = cdda_identify(devname.constData(), 0, NULL);
 
@@ -172,7 +177,7 @@
     return 0;
 }
 
-#ifdef HAVE_CDIO
+#if defined(HAVE_CDIO) || defined(HAVE_CDIO_PARANOIA)
 static void paranoia_cb(long, paranoia_cb_mode_t)
 {
 }
@@ -413,7 +418,7 @@
 
 int CDRipperThread::ripTrack(QString &cddevice, Encoder *encoder, int tracknum)
 {
-#ifdef HAVE_CDIO
+#if defined(HAVE_CDIO) || defined(HAVE_CDIO_PARANOIA)
     QByteArray devname = cddevice.toLatin1();
     cdrom_drive *device = cdda_identify(devname.constData(), 0, NULL);
 
@@ -873,7 +878,7 @@
 
 void Ripper::scanCD(void)
 {
-#ifdef HAVE_CDIO
+#if defined(HAVE_CDIO) || defined(HAVE_CDIO_PARANOIA)
     {
     LOG(VB_MEDIA, LOG_INFO, QString("Ripper::%1 CD='%2'").
         arg(__func__).arg(m_CDdevice));
@@ -1226,7 +1231,7 @@
     bool bEjectCD = gCoreContext->GetNumSetting("EjectCDAfterRipping",1);
     if (bEjectCD)
     {
-#ifdef HAVE_CDIO
+#if defined(HAVE_CDIO) || defined(HAVE_CDIO_PARANOIA)
         LOG(VB_MEDIA, LOG_INFO, QString("Ripper::%1 '%2'").
             arg(__func__).arg(m_CDdevice));
         (void)cdio_eject_media_drive(m_CDdevice.toLatin1().constData());
diff -ruN mythtv-0.28/mythplugins/mythmusic/mythmusic/decoder.cpp mythtv-0.28_new/mythplugins/mythmusic/mythmusic/decoder.cpp
--- mythtv-0.28/mythplugins/mythmusic/mythmusic/decoder.cpp	2016-04-11 20:30:28.000000000 +0100
+++ mythtv-0.28_new/mythplugins/mythmusic/mythmusic/decoder.cpp	2016-11-12 17:12:58.338077051 +0000
@@ -71,7 +71,7 @@
     {
         factories = new QList<DecoderFactory*>;
 
-#ifdef HAVE_CDIO
+#if defined(HAVE_CDIO) || defined(HAVE_CDIO_PARANOIA)
         Decoder::registerFactory(new CdDecoderFactory);
 #endif
         Decoder::registerFactory(new avfDecoderFactory);
diff -ruN mythtv-0.28/mythplugins/mythmusic/mythmusic/main.cpp mythtv-0.28_new/mythplugins/mythmusic/mythmusic/main.cpp
--- mythtv-0.28/mythplugins/mythmusic/mythmusic/main.cpp	2016-04-11 20:30:28.000000000 +0100
+++ mythtv-0.28_new/mythplugins/mythmusic/mythmusic/main.cpp	2016-11-12 17:12:58.332077155 +0000
@@ -47,7 +47,7 @@
 #include "importmusic.h"
 #include "metaio.h"
 
-#ifdef HAVE_CDIO
+#if defined(HAVE_CDIO) || defined(HAVE_CDIO_PARANOIA)
 #include "cdrip.h"
 #endif
 
@@ -638,7 +638,7 @@
     }
 }
 
-#ifdef HAVE_CDIO
+#if defined(HAVE_CDIO) || defined(HAVE_CDIO_PARANOIA)
 static void handleCDMedia(MythMediaDevice *cd)
 {
 
diff -ruN mythtv-0.28/mythplugins/mythmusic/mythmusic/musicplayer.cpp mythtv-0.28_new/mythplugins/mythmusic/mythmusic/musicplayer.cpp
--- mythtv-0.28/mythplugins/mythmusic/mythmusic/musicplayer.cpp	2016-04-11 20:30:28.000000000 +0100
+++ mythtv-0.28_new/mythplugins/mythmusic/mythmusic/musicplayer.cpp	2016-11-12 17:12:58.335077103 +0000
@@ -24,7 +24,7 @@
 #include "decoder.h"
 #include "decoderhandler.h"
 #include "metaio.h"
-#ifdef HAVE_CDIO
+#if defined(HAVE_CDIO) || defined(HAVE_CDIO_PARANOIA)
 #include "cddecoder.h"
 #endif
 #include "constants.h"
@@ -1548,7 +1548,7 @@
     LOG(VB_PLAYBACK, LOG_INFO, QString ("decoder handler is ready, decoding %1")
             .arg(getDecoder()->getURL()));
 
-#ifdef HAVE_CDIO
+#if defined(HAVE_CDIO) || defined(HAVE_CDIO_PARANOIA)
     CdDecoder *cddecoder = dynamic_cast<CdDecoder*>(getDecoder());
     if (cddecoder)
         cddecoder->setDevice(gCDdevice);
