--- configure	2016-11-12 12:22:23.065867667 +0000
+++ configure_new	2016-11-12 12:21:21.052900826 +0000
@@ -682,7 +682,7 @@
         echo "MythMusic requires FLAC."
     fi
 
-    if ! check_lib cdio/cdio.h cdio_open -lcdio || ! check_lib cdio/cdda.h cdio_cddap_open -lcdio_cdda || ! check_lib cdio/paranoia.h cdio_paranoia_init -lcdio_paranoia ; then
+    if ! check_lib cdio/cdio.h cdio_open -lcdio || (! check_lib cdio/cdda.h cdio_cddap_open -lcdio_cdda && ! check_lib cdio/paranoia/cdda.h cdio_cddap_open -lcdio_cdda) || (! check_lib cdio/paranoia.h cdio_paranoia_init -lcdio_paranoia && ! check_lib cdio/paranoia/paranoia.h cdio_paranoia_init -lcdio_paranoia) ; then
         disable cdio
     fi
 
